crates/polars-core/src/chunked_array/ops/explode_and_offsets.rs

```diff
@@ -66,7 +69,10 @@ impl ChunkExplode for ListChunked {
         let offsets = listarr.offsets().as_slice();
         let mut values = listarr.values().clone();
 
-        let (mut s, offsets) = if ca._can_fast_explode() {
+        let (mut s, offsets) = if ca._can_fast_explode()
+            && (!options.empty_as_null || !ca.has_empty_lists())
+            && (!options.keep_nulls || !ca.has_nulls())
```

Comment:
Since `has_empty_lists` can be quite a bit more expensive than `has_nulls`, can you reorder these two conditions to first check the `keep_nulls`/`has_nulls` condition?